metagraph/tests/test_vector_algorithm.cpp

```diff
@@ -160,6 +160,172 @@ TEST(IntVector, call_nonzeros_sparse_every_4) {
+        for (auto memorder : { __ATOMIC_RELAXED, __ATOMIC_SEQ_CST }) {
+            sdsl::int_vector<> vector_atomic = aligned_int_vector(600, 0, w, 16);
+            uint64_t val = sdsl::bits::lo_set[w];
+
+            std::mutex mu;
+            std::atomic_thread_fence(std::memory_order_release);
+            #pragma omp parallel for num_threads(3) schedule(static, 1)
+            for (size_t j = 0; j < w; ++j) {
+                uint64_t set_val = 1llu << j;
+                for (size_t i = 0; i < vector_atomic.size(); ++i) {
+                    EXPECT_EQ(0u, atomic_fetch_and_add(vector_atomic, i, set_val,
+                                                       mu, memorder) & set_val);
+
+                    // some added contention
+                    atomic_fetch_and_add(vector_atomic, 0, 0, mu, memorder);
+
+                    size_t r = 5; // this test is faster, so we can make this wider
+                    size_t i_min = i - std::min(i, r);
+                    size_t i_max = std::min(i + r, size_t(vector_atomic.size()));
+                    for (size_t k = i_min; k < i_max; ++k) {
+                        atomic_fetch_and_add(vector_atomic, k, 0, mu, memorder);
+                    }
+                }
+            }
+
+            std::atomic_thread_fence(std::memory_order_acquire);
+            EXPECT_EQ(sdsl::int_vector<>(600, val, w), vector_atomic);
+        }
+    }
+}
+
+TEST(IntVector, atomic_fetch_and_add_all_bits_except_one) {
+    for (size_t w = 1; w <= 64; ++w) {
+        for (auto memorder : { __ATOMIC_RELAXED, __ATOMIC_SEQ_CST }) {
+            for (size_t s = 0; s < w; s += 3) {
+                sdsl::int_vector<> vector_atomic = aligned_int_vector(600, 0, w, 16);
+                uint64_t val = sdsl::bits::lo_set[w] ^ (1llu << s);
+
+                std::mutex mu;
+                std::atomic_thread_fence(std::memory_order_release);
+                #pragma omp parallel for num_threads(3) schedule(static, 1)
+                for (size_t j = 0; j < w; ++j) {
+                    if (j != s) {
+                        uint64_t set_val = 1llu << j;
+                        for (size_t i = 0; i < vector_atomic.size(); ++i) {
+                            EXPECT_EQ(0u, atomic_fetch_and_add(vector_atomic, i, set_val,
+                                                               mu, memorder) & set_val);
+
+                            // some added contention
+                            atomic_fetch_and_add(vector_atomic, 0, 0, mu, memorder);
+
+                            size_t r = 2;
+                            size_t i_min = i - std::min(i, r);
+                            size_t i_max = std::min(i + r, size_t(vector_atomic.size()));
+                            for (size_t k = i_min; k < i_max; ++k) {
+                                atomic_fetch_and_add(vector_atomic, k, 0, mu, memorder);
+                            }
+                        }
+                    }
+                }
```

Comment:
if this works with __ATOMIC_RELAXED, why test others?

Reply:
Originally I had it set up so that if any checks failed, it would output the memory order and I'd know if the issue was due to memory order or something else. Though in the future if issues do come up, one could just modify this test to output the memory order.